### DISCUSSION TITLE
Why does subscription create Stripe customer but not one time payment?

### DISCUSSION THREAD

**Discussion by benf2004** [Glench/ExtPay#39] [Q&A]
Hello everyone & @Glench. 

I am working on integrating my extension with an affiliate tracking software called Rewardful. In order for this to function properly, I pass an email from my extension and Rewardful matches it with a Stripe customer.  However, ExtPay only creates a Stripe customer when they choose a subscription plan. When I create a one-time payment, it is not attached to a Stripe customer & the payment is considered a "[guest](https://support.stripe.com/questions/guest-customer-faq)". I would like to _always_ create a Stripe customer so this affiliate tracking works successfully. Is there a way to make this happen?

Thanks!

**Comment by benf2004** [comment]
Update: I have switched to a different referral program called LeadDyno which works without having to create a Stripe customer object. Still would be an interesting idea though.

**Comment by ianustec** [comment]
HI,

thank you for your work.

I have the same problem. I need to _always_ create a Customer Stripe with its ID and not just the guest. Possible?
